wagtail_localize/models.py

```diff
@@ -2191,14 +2191,59 @@ def disable_translation_on_delete(instance, **kwargs):
     )
 
 
+def cleanup_translation_on_delete(instance, **kwargs):
+    """
+    When either the source or destination object is deleted, remove the corresponding translation data.
+
+    When all translations of the object are removed, delete all remaining metadata.
+    """
+    translation_key = instance.translation_key
+    locale_id = instance.locale_id
+
+    StringTranslation.objects.filter(
+        context__object_id=translation_key, locale=locale_id
+    ).delete()
+    SegmentOverride.objects.filter(
+        context__object__translation_key=translation_key, locale=locale_id
+    ).delete()
+    Translation.objects.filter(source__object_id=translation_key).filter(
+        # Remove the translations where this object was the source
+        Q(source__locale_id=instance.locale_id)
+        # Removed the translations where this object was the destination
+        | Q(target_locale_id=instance.locale_id)
```

Comment:
Can you add some context why we are using `instand.locale_id` instead of `locale_id` here?

Reply:
good spot. did a refactor and missed those